kong/clustering/init.lua

```diff
@@ -215,13 +271,36 @@ function _M:init_worker()
     return { name = p.name, version = p.handler.VERSION, }
   end, self.plugins_list)
 
-  for _, ch in ipairs{"child", "json_handler", "wrpc_handler"} do
-    local child = self[ch]
-    if child then
-      child:init_worker()
-    end
+  local role = self.conf.role
+  if role == "control_plane" then
+    self.json_handler:init_worker()
+    self.wrpc_handler:init_worker()
+  end
+
+  if role == "data_plane" and ngx.worker.id() == 0 then
+    assert(ngx.timer.at(0, function(premature)
+      if premature then
+        return
+      end
+
+      self:request_version_negotiation()
+
+      local config_proto = version_negotiation.get_negotiated_service("config")
+      ngx_log(ngx_DEBUG, _log_prefix, "config_proto: ", config_proto)
+      if config_proto == "v1" then
+        self.child = require "kong.clustering.wrpc_data_plane".new(self)
+
+      elseif config_proto == "v0" or config_proto == nil then
+        self.child = require "kong.clustering.data_plane".new(self)
+
+      end
```

Comment:
Nice to see negotiation actually happening.
I think having a cluster_protocol configuration value in kong.conf that is interpreted only by the CP would be helpful.
The default value will be empty, which translates to "use whatever the code is doing, do not influence it).
This configuration option could be used as an escape hatch in the case where wRPC implementation has a serious bug. This should be done in a follow up PR. What do you think?

Reply:
i'm thinking of a more generic "killswitch", that simply removes options from the requested services.  something like `banned=metrics, config.v0` would remove the whole `metrics` service, and the option `v0` from `config`.